RunningOrder/Shared/AppStateManager.swift

```diff
@@ -20,16 +20,60 @@ extension AppStateManager {
     }
 }
 
-extension AppStateManager {
-    enum Error: Swift.Error {
-        case unexistingError // in the mapError I need a concrete error to return, but I don't want to use an existing one in order to identify a potential bug if this error is really thrown
-        case fetchTimeout
+struct ErrorTrace {
+    struct Location: CustomDebugStringConvertible {
+        let file: String
+        let line: Int
+        let function: String
+
+        static func collect(file: String = #fileID, line: Int = #line, function: String = #function) -> Location {
```

Comment:
What does the "#" ?

Reply:
nothing special, it is part of the aliases : 
<img width="781" alt="image" src="https://user-images.githubusercontent.com/6250452/126762448-d93b1d40-29d0-42de-9aed-4ca4737e2290.png">
More info at : https://docs.swift.org/swift-book/ReferenceManual/Expressions.html